### PR TITLE
Combat Tentacles

### PR DESCRIPTION
**Pull request by tigercat2000**
tgstation/tgstation#22203

Combat tentacles! Changelings can now buy tactical tentacle action for 2 evolution points.

The tentacle behaves differently based on intent:
 - Help intent drags people and items closer. You will automatically grab the latter.
 - Disarm intent grabs things out of people's active hands.
 - Grab intent puts the target into an aggressive grab as soon as they are within range.
 - Harm intent will impale the target if you have a sharp object in your hand, or just drag them to you otherwise. Impaling is pretty much just a 1 hit automatic attack doing no damage, but it looks cool!

:cl:
add: Changelings are now tentacle monsters. Give them a schoolgirl outfit.
/:cl:

### CODE REVIEW

**Comment by Xantholne** [comment]
WTF indeed.

**Comment by SamHPurp** [comment]
This is perfect ~~for ERP~~ for catching those that flee my armblade...

**Comment by Kyep** [comment]
Sure.

Also, please port horror form.

**Comment by deep-fried-faceplant** [comment]
GET OVER HERE!

Oh my god, I love this. The pull and stun might be a little too much though, I would love to see this feature be a combat one and not a stun one. Perhaps have the harm intent just be a ranged melee attack?

**Comment by Spacemanspark** [comment]
Come to pudge

**Comment by granodd** [comment]
I can imagine the shenanigans with this.

"This is _mine_ now."

**Comment by KingPhilipIII** [comment]
They changed the tentacle stun time to a weaken I believe. It's actually an incredibly robust additon to the armblade.

**Comment by Fox-McCloud** [comment]
>Come to pudge

We already have a meathook...

**Comment by Spacemanspark** [comment]
I made the joke on that PR too but nobody got it

File: code/game/gamemodes/changeling/powers/mutations.dm [hunk 1]
```diff
@@ -174,6 +177,180 @@
...
+	if(H.in_throw_mode)
+		H.throw_mode_off() //Don't annoy the changeling if he doesn't catch the item
+
+/mob/proc/tentacle_grab(mob/living/carbon/C)
+	if(Adjacent(C))
+		var/obj/item/weapon/grab/G = C.grabbedby(src,1)
+		if(istype(G))
+			G.state = GRAB_AGGRESSIVE //Instant aggressive grab
+
+/mob/proc/tentacle_stab(mob/living/carbon/C)
```
Comment (Fox-McCloud) [inline code/game/gamemodes/changeling/powers/mutations.dm]: mob proc instead of a gun proc?

Comment (tigercat2000) [inline code/game/gamemodes/changeling/powers/mutations.dm]: I tried everything to make the projectile procs work, but they literally never fired regardless of how i setup the callback, addtimer, spawned call, the projectile always managed to get deleted beforehand. This was the easiest solution.

Comment (Fox-McCloud) [inline code/game/gamemodes/changeling/powers/mutations.dm]: ahhhh, so this is what you were referring to.

Sounds like something, somewhere, need to be updated in projectile code.

**Comment by KasparoVy** [comment]
I can't wait to get obscenely fucked over by this.

**Comment by scrubmcnoob** [comment]
where's the original japanese version for this PR?
i don't watch dubs

On a real note, be interesting seeing changelings get this. Add a ton more to their 'monster' feel and how uncrew they really are. Also horror form when.

**Comment by Rapidvalj** [comment]
horror form on top of this and I can die a happy man

File: code/game/gamemodes/changeling/powers/mutations.dm [hunk 2]
```diff
@@ -174,6 +177,180 @@
...
+	if(Adjacent(C))
+		var/obj/item/I = r_hand
+		if(!is_sharp(I))
+			I = l_hand
+		if(!is_sharp(I))
+			C.visible_message("<span class='danger'>[C] falls at [src]'s feet!</span>", "<span class='userdanger'>You are thrown at [src]'s feet!</span>")
+			C.Weaken(2)
```
Comment (Fox-McCloud) [inline code/game/gamemodes/changeling/powers/mutations.dm]: I believe the two lines above can be removed; they got ride of weakening for this ability as it was deemed too powerful: https://github.com/tgstation/tgstation/blob/1c8badd8fef34ab8ecd3729b01a2bccf01de4db6/code/game/gamemodes/changeling/powers/mutations.dm#L302

Unless I"m missing something.

**Comment by GutTC** [comment]
And when you thought you could get away running in a one tile thin maintenance walkway...

But holy. The disarm function looks real spooky tho'.

**Pull request merged**